### PR TITLE
Military Base location

### PR DESCRIPTION
**Pull request by RarkGrames**
#### Summary
```SUMMARY: Content "Added a military base location"```

#### Purpose of change
- Adding a new 10x13 military base location.
- Adding a new scenario (Overrun) to go with it. A soldier got stuck in the base that was overrun by a horde.

#### Describe the solution

Refugee camp in the front parking lot

Corpse burning pits

Entrance

One of the barracks

Canteen

Main warehouse

Comm center

Back entrance

Armory

Garage

Maintenance, utilities, storage

Improvised field hospital

Infirmary north part

Infirmary south part

Storage and vehicle refilling area in the back of the base

Command center offices

Command center underground

Bird's eye view northwest

Northeast

West

East

Southwest

Southeast

Tunnels

#### Testing
A lot.

#### Additional context
Additional stuff that had to be created for it to work:
- Bunk bed furniture
- Warehouse shelf furniture
- Mess tray item
- IV bag for blood and saline
- Some existing item group changes

Feedback or even testing is welcome.

### CODE REVIEW

**Comment by CountAlex** [comment]
That's pretty awesome! And new scenario too.
I would suggest using concrete and yellow striped concrete for helipad on one of the screenshots. Plus I think it should be bigger if there's a chance to spawn helicopters on it (eventually, maybe). Current size is really tight even for UH-60 which is smallest of the copters existing in game.

File: data/json/scenarios.json [hunk 1]
```diff
@@ -491,5 +491,16 @@
...
+  },
+  {
+    "type": "scenario",
+    "ident": "overrun",
+    "name": "Overrun",
+    "points": 4,
+    "description": "When your base got overrun by the dead, your last order was to retreat to the armory.  During all the chaos you got separated from your squad and you are now stuck in the warehose all alone.  You are not sure if anyone made it to the armory, or you are the last man alive.",
+    "start_name": "Military Base Warehouse",
+    "allowed_locs": [ "mil_base_2g" ],
+    "professions": [ "soldier" ],
```
Comment (LaVeyanFiend) [inline data/json/scenarios.json]: Not at least allowing the same military professions as allowed in Helicopter crash?

**Comment by I-am-Erk** [comment]
This looks fantastic...this is way too much for a single PR though. For a start I'd suggest peeling the furniture out to its own PR and tossing it into a couple other maps where it makes sense. I'll look closer to see what else could be broken into its own thing too. It's not a great idea to submit five thousand lines in a single batch, we aim for a cap of about 1000 per PR.

**Comment by curstwist** [comment]
Looks great. there is a `f_shredder` you can add to the office areas too.  I'll take a closer look soon for mapgen stuff.

**Comment by 1n17** [comment]
Pretty awesome, but wouldn't such a large, permanent military installation warrant some more underground development? Perhaps even connection to lab subway network

**Comment by RarkGrames** [comment]
> 
> 
> That's pretty awesome! And new scenario too.
> I would suggest using concrete and yellow striped concrete for helipad on one of the screenshots. Plus I think it should be bigger if there's a chance to spawn helicopters on it (eventually, maybe). Current size is really tight even for UH-60 which is smallest of the copters existing in game.

It is intended for an UH-60, and researching the helipad minimal sizes I've [found ](https://www.tc.gc.ca/en/transport-canada/corporate/acts-regulations/regulations/sor-96-433/part3-standards-325-325-160.htm) this one: 

"The specifications in this Standard are based on a FATO size that is 1.5 times larger than the longest helicopter for which the heliport is certified."

This one has at least 19 tiles of space until the nearest solid object, and the UH-60 is 13 tiles long, so theoretically it would fit.

I will see if I can rearrange things to make it large enough for the VTOL plane or a Chinook, but I doubt it.

I can change the pavement if I haven't ran out of available characters in the mapgen palette.

File: data/json/scenarios.json (hunk 1 above)
Comment (RarkGrames) [reply data/json/scenarios.json]: Added specops, bio_soldier and bio_sniper to starting professions.

**Comment by RarkGrames** [comment]
@I-am-Erk I can do that with the furniture and the items. Next would be like the empty building, or separate monster/item groups?

**Comment by RarkGrames** [comment]
> 
> 
> Pretty awesome, but wouldn't such a large, permanent military installation warrant some more underground development? Perhaps even connection to lab subway network

Do you have any specific facilities in mind that can be added as underground sections?

**Comment by RarkGrames** [comment]
> 
> 
> Looks great. there is a `f_shredder` you can add to the office areas too. I'll take a closer look soon for mapgen stuff.

Added shredders to offices

**Comment by mrkybe** [comment]
> > Pretty awesome, but wouldn't such a large, permanent military installation warrant some more underground development? Perhaps even connection to lab subway network
> 
> Do you have any specific facilities in mind that can be added as underground sections?

I don't know if underground necessarily makes sense, but a laundry room wouldn't be out of place for such a large base.

**Comment by RarkGrames** [comment]
Yeah, towards the end I realized that there is no laundry room, but it was too late to stick it in somewhere. But placing it underground will work.

**Comment by Amoebka** [comment]
Does this have any random/nested mapgen at all or is it entirely static? Being a huge static location was one of the main drawbacks of the National Guard Camp.

**Comment by RarkGrames** [comment]
It is completely static.

**Comment by Brian-Otten** [comment]
I hope this makes it in and give it 2 thumbs up, would love to see this while i play.

**Comment by I-am-Erk** [comment]
It's okay to leave this static for now, but we should probably discuss a second pass adding some variable maps, nested mapgen and map extras when it's done... mainly because it'd be awesome.

@1n17 connecting every military base to top secret high-security underground labs is all that great. Eventually when we have variable mapgen we could make a version of this that has a subway connection.

**Comment by RarkGrames** [comment]
> 
> 
> It's okay to leave this static for now, but we should probably discuss a second pass adding some variable maps, nested mapgen and map extras when it's done... mainly because it'd be awesome.
> 
> @1n17 connecting every military base to top secret high-security underground labs is all that great. Eventually when we have variable mapgen we could make a version of this that has a subway connection.

I was planning to add randomized stuff like wrecks, bodies and casings for fluff. The main part that peope would visit for (loot) would stay static anyway, and fluff becomes irrelevant after the third visit to the location.

**Comment by RarkGrames** [comment]
Added laundry room.

Untitled

**Comment by RarkGrames** [comment]
Added some chairs to the laundry (completely ran out of characters for adding bench).

**Comment by RarkGrames** [comment]
Rearranged some things to make the helipad larger.

**Comment by Night-Pryanik** [comment]
> completely ran out of characters for adding bench

You can override any character from palette. Just add furniture `f_bench` and respective terrain in `mil_base_tunnels_i` and assign it for any unassigned (in this particular overmap terrain) symbol, `B` for example. So, while `B` is used for `f_tourist_table` in palette, it will be used for `f_bench` only in `mil_base_tunnels_i` OM terrain.

**Comment by RarkGrames** [comment]
Ok, that worked.

**Comment by RarkGrames** [comment]
Ok, that worked.

File: data/json/overmap/overmap_special/specials.json [hunk 2]
```diff
@@ -6461,5 +6461,249 @@
...
+      { "point": [ 7, 10, 1 ], "overmap": "mil_base_8k1_north" },
+      { "point": [ 2, 8, 2 ], "overmap": "mil_base_3i2_north" },
+      { "point": [ 2, 8, 3 ], "overmap": "mil_base_3i3_north" },
+      { "point": [ 2, 8, 4 ], "overmap": "mil_base_3i4_north" }
+    ],
+    "locations": [ "field" ],
+    "connections": [ { "point": [ 3, -2, 0 ] } ],
+    "city_distance": [ 4, -1 ],
+    "city_sizes": [ 4, 12 ],
+    "occurrences": [ 0, 1 ],
```
Comment (I-am-Erk) [inline data/json/overmap/overmap_special/specials.json]:
```suggestion
    "occurrences": [ 1, 10 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 3]
```diff
@@ -6461,5 +6461,249 @@
...
+    "flags": [ "CLASSIC" ]
```
Comment (I-am-Erk) [inline data/json/overmap/overmap_special/specials.json]:
```suggestion
    "flags": [ "UNIQUE", "MILITARY" ]
```
I'd suggest making this much rarer than every 1 in 2 overmaps. The unique flag lets you tweak it more precisely, I've written it here for one in ten overmaps.

File: data/json/overmap/overmap_special/specials.json (hunk 2 above)
Comment (RarkGrames) [reply data/json/overmap/overmap_special/specials.json]: Wouldn't this make it that the base spawns on every overmap 1-10 times?

Comment (Night-Pryanik) [reply data/json/overmap/overmap_special/specials.json]: > I'm not sure how "occurrences" interacts with the "unique" flag.

`"occurrences": [ X, Y ]`
With `UNIQUE` flag, the chance to spawn location will be the X out of Y per overmap. Without this flag, the location will spawn minimum X and maximum Y times per overmap.

Comment (I-am-Erk) [reply data/json/overmap/overmap_special/specials.json]: The military flag has no impact on the 1,10 thing, I'm just encouraging more detailed flagging of om specials for upcoming biome changes.

File: data/json/overmap/overmap_special/specials.json (hunk 3 above)
Comment (curstwist) [reply data/json/overmap/overmap_special/specials.json]: make sure you don't set occurrences 1st value to 0 if you use the unique flag. Or you'll end up with 0% chance in x to spawn as night outlined.  Did that with the isherwood special and had to fix it.

**Comment by RarkGrames** [comment]
Ok, that's clear now. Thanks for the explanations.

**Pull request merged**